WordPress/Sniffs/PHP/NoSilencedErrorsSniff.php

```diff
@@ -0,0 +1,208 @@
+		'file_get_contents'     => true,
+		'file'                  => true,
+		'fileatime'             => true,
+		'filectime'             => true,
+		'filegroup'             => true,
+		'fileinode'             => true,
+		'filemtime'             => true,
+		'fileowner'             => true,
+		'fileperms'             => true,
+		'filesize'              => true,
+		'filetype'              => true,
+		'fopen'                 => true,
+		'is_dir'                => true,
+		'is_executable'         => true,
+		'is_file'               => true,
+		'is_link'               => true,
+		'is_readable'           => true,
+		'is_writable'           => true,
+		'is_writeable'          => true,
+		'lstat'                 => true,
+		'mkdir'                 => true,
+		'move_uploaded_file'    => true,
+		'readfile'              => true,
+		'readlink'              => true,
+		'rename'                => true,
+		'rmdir'                 => true,
+		'stat'                  => true,
+		'unlink'                => true,
+
+		// FTP extension.
+		'ftp_chdir'             => true,
+		'ftp_login'             => true,
+		'ftp_rename'            => true,
+
+		// Stream extension.
+		'stream_select'         => true,
+		'stream_set_chunk_size' => true,
+
+		// Zlib extension.
+		'deflate_add'           => true,
+		'deflate_init'          => true,
+		'inflate_add'           => true,
+		'inflate_init'          => true,
+		'readgzfile'            => true,
+
+		// Miscellaneous other functions.
+		'imagecreatefromstring' => true,
+		'parse_url'             => true, // Pre-PHP 5.3.3 an E_WARNING was thrown when URL parsing failed.
+		'unserialize'           => true,
+	);
+
+	/**
+	 * Tokens which are regarded as empty for the purpose of determining
+	 * the name of the called function.
+	 *
+	 * This property is set from within the register() method.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @var array
+	 */
+	private $empty_tokens = array();
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array
+	 */
+	public function register() {
+		$this->empty_tokens                    = Tokens::$emptyTokens;
+		$this->empty_tokens[ \T_NS_SEPARATOR ] = \T_NS_SEPARATOR;
+		$this->empty_tokens[ \T_BITWISE_AND ]  = \T_BITWISE_AND;
+
+		return array(
+			\T_ASPERAND,
+		);
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 */
+	public function process_token( $stackPtr ) {
+		// Handle the user-defined custom function whitelist.
+		$this->custom_whitelist = $this->merge_custom_array( $this->custom_whitelist, array(), false );
+		$this->custom_whitelist = array_map( 'strtolower', $this->custom_whitelist );
+
+		/*
+		 * Check if the error silencing if done for one of the whitelisted functions.
+		 */
+		$nextNonEmpty = $this->phpcsFile->findNext( $this->empty_tokens, ( $stackPtr + 1 ), null, true, null, true );
```

Comment:
Shouldn't this variable name be underscored to comply with WP CS?

Reply:
Fixed, here and in a few other places.

It's funny, I tend to use snake_case for everything, until I name a variable holding the return value of a pascalCase PHPCS native function call,.. :confused: